### PR TITLE
Fix support for Laravel 11

### PR DESCRIPTION
**Pull request by nhedger**
- [ ] Added or updated tests
- [ ] Documented user facing changes
- [ ] Updated CHANGELOG.md

**Changes**

WIP

**Breaking changes**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -82,5 +82,10 @@ parameters:
       - '#PHPDoc tag @var for variable \$router contains unknown class Laravel\\Lumen\\Routing\\Router\.#'
       - '#Call to method addRoute\(\) on an unknown class Laravel\\Lumen\\Routing\\Router\.#'
 
+  # Recent Mockery version do not account for the last arg possibly being a closure
+  - path: tests/Unit/Subscriptions/Iterators/AuthenticatingSyncIteratorTest.php
+    message: '#Parameter \#2 \.\.\.\$args of static method Mockery::mock\(\) expects array\|class-string<Nuwave\\Lighthouse\\Subscriptions\\SubscriptionGuard>\|Nuwave\\Lighthouse\\Subscriptions\\SubscriptionGuard, Closure given\.#'
```
Comment (nhedger) [inline phpstan.neon:87]: Looks like the type information does not handle the last argument possibly being a Closure.

File: tests/Unit/Schema/Directives/DeprecatedDirectiveTest.php [hunk 2]
```diff
@@ -58,7 +58,7 @@ enumValues(includeDeprecated: $includeDeprecated) {
 
         $types = $withoutDeprecatedIntrospection->json('data.__schema.types');
         $foo = Arr::first($types, static fn (array $type): bool => $type['name'] === 'Foo');
-        $this->assertCount(1, $foo['enumValues']);
```
Comment (nhedger) [inline tests/Unit/Schema/Directives/DeprecatedDirectiveTest.php:61]: Alternatively, we could just ignore the error if you don't want the added noise in the test

File: phpstan.neon [hunk 3]
```diff
@@ -82,5 +82,10 @@ parameters:
       - '#PHPDoc tag @var for variable \$router contains unknown class Laravel\\Lumen\\Routing\\Router\.#'
       - '#Call to method addRoute\(\) on an unknown class Laravel\\Lumen\\Routing\\Router\.#'
 
+  # Recent Mockery version do not account for the last arg possibly being a closure
```
Comment (spawnia) [inline phpstan.neon:85]:
```suggestion
  # Recent Mockery versions do not account for the last arg possibly being a closure
```

File: tests/Integration/Schema/Types/InterfaceTest.php [hunk 4]
```diff
@@ -341,11 +341,11 @@ interface HasPosts {
...
     }
 
-    /** @return \Illuminate\Database\Eloquent\Collection<int, \Tests\Utils\Models\User|\Tests\Utils\Models\Team> */
-    public static function fetchResults(): EloquentCollection
+    /** @return \Illuminate\Support\Collection<int, \Tests\Utils\Models\User|\Tests\Utils\Models\Team> */
+    public static function fetchResults(): Collection
     {
-        /** @var \Illuminate\Database\Eloquent\Collection<int, \Tests\Utils\Models\User|\Tests\Utils\Models\Team> $results */
-        $results = new EloquentCollection();
+        /** @var \Illuminate\Support\Collection<int, \Tests\Utils\Models\User|\Tests\Utils\Models\Team> $results */
```
Comment (spawnia) [inline tests/Integration/Schema/Types/InterfaceTest.php:347]: Yeah, this is better. Eloquent collections are supposed to hold only a single model type.

**Comment by spawnia** [comment]
Looks good so far. Thank you for tackling this.

File: phpstan.neon [hunk 5]
```diff
@@ -82,5 +82,16 @@ parameters:
...
+    message: '#Parameter \#2 \.\.\.\$args of static method Mockery::mock\(\) expects array\|class-string<Nuwave\\Lighthouse\\Subscriptions\\SubscriptionGuard>\|Nuwave\\Lighthouse\\Subscriptions\\SubscriptionGuard, Closure given\.#'
 
+  - path: tests/Integration/Schema/Directives/BuilderDirectiveTest.php
+    message: '#Return type of call to static method Mockery::mock\(\) contains unresolvable type\.#'
+
+  # Laravel added generics to Arr::first in 11.x, not sure how to handle them for now.
+  - path: src/OrderBy/OrderByDirective.php
+    message: '#Unable to resolve the template type TValue in call to method static method Illuminate\\Support\\Arr::first\(\)#'
+  - path: src/Testing/TestResponseMixin.php
+    message: '#Unable to resolve the template type TValue in call to method static method Illuminate\\Support\\Arr::first\(\)#'
```
Comment (nhedger) [inline phpstan.neon:96]: This one is a PITA to debug. I think it's fine to silence for now if tests pass. Let me know if you're good with it.

**Comment by spawnia** [comment]
> Some tests are failing with the following error:
> 
> ```
> Error: Child process error (exit code 1): ErrorException thrown in /home/runner/work/lighthouse/lighthouse/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php on line 335 while loading bootstrap file /home/runner/work/lighthouse/lighthouse/vendor/larastan/larastan/bootstrap.php: symlink(): File exists
> 
>  -- ----------------------------------------------------------------------------------------------------------- 
>      Error                                                                                                      
>  -- ----------------------------------------------------------------------------------------------------------- 
>      Child process error (exit code 1): ErrorException thrown in                                                
>      /home/runner/work/lighthouse/lighthouse/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php  
>      on line 335 while loading bootstrap file                                                                   
>      /home/runner/work/lighthouse/lighthouse/vendor/larastan/larastan/bootstrap.php:                            
>      symlink(): File exists                                                                                     
>                                                                                                                 
>  -- -----------------------------------------------------------------------------------------------------------
> ```
> 
> I don't think it's related to the upgrade, but rather an issue with the CI perhaps ?

Yeah, I just do not get why that one always fails.

File: tests/Unit/Subscriptions/Iterators/AuthenticatingSyncIteratorTest.php [hunk 6]
```diff
@@ -1,4 +1,6 @@
-<?php declare(strict_types=1);
+<?php
+
+declare(strict_types=1);
```
Comment (spawnia) [inline tests/Unit/Subscriptions/Iterators/AuthenticatingSyncIteratorTest.php:3]:
```suggestion
<?php declare(strict_types=1);
```

File: tests/Unit/Subscriptions/Iterators/AuthenticatingSyncIteratorTest.php [hunk 7]
```diff
@@ -79,7 +81,9 @@ static function (Subscriber $subscriber) use (&$processedItems, &$authenticatedU
 
 final class AuthenticatingSyncIteratorAuthenticatableStub implements Authenticatable
 {
-    public function __construct(private int $id) {}
+    public function __construct(private int $id)
+    {
+    }
```
Comment (spawnia) [inline tests/Unit/Subscriptions/Iterators/AuthenticatingSyncIteratorTest.php:86]:
```suggestion
    public function __construct(private int $id) {}
```

**Comment by spawnia** [comment]
Thanks, nicely done!

**Pull request merged**